lib/common/monitoring/alarms/UsageAlarmFactory.ts

```diff
@@ -60,10 +60,14 @@ export class UsageAlarmFactory {
     percentMetric: MetricWithAlarmSupport,
     props: UsageThreshold,
     disambiguator?: string,
-    usageType?: UsageType
+    usageType?: UsageType,
+    additionalAlarmNameSuffix?: string
   ) {
-    const alarmNameSuffix: string =
-      usageType === undefined ? "CPU-Usage" : `${usageType}-CPU-Usage`;
+    const alarmNameSuffix: string = [
+      usageType,
+      "CPU-Usage",
+      additionalAlarmNameSuffix,
+    ].join("-");
```

Comment:
You should filter out the null/undefined items first to retain backwards compatibility (see [failed test snapshot diffs](https://github.com/cdklabs/cdk-monitoring-constructs/actions/runs/6580538796/job/17902046124#step:8:676)).

Reply:
done